ton_client/src/abi/encode_message.rs

```diff
@@ -568,30 +583,45 @@ pub async  fn encode_internal_message(
                 bounce,
             )?
         } else {
-            encode_empty_int_deploy(image, workchain_id, ihr_disabled, bounce)?
+            encode_empty_int_deploy(src_address, image, workchain_id, ihr_disabled, bounce)?
         }
-    } else if let Some(call_set) = &params.call_set {
+    } else {
         let address = params
             .address
             .as_ref()
             .ok_or(abi::Error::required_address_missing_for_encode_message())?;
         let address = account_decode(address)?;
+        let value = CurrencyCollection::with_grams(
+            u64::from_str(&params.value)
+                .map_err(|err| abi::Error::encode_run_message_failed(err, ""))?
+        );
+        if let Some(call_set) = &params.call_set {
+            let abi = params.abi
+                .ok_or_else(|| Error::invalid_abi("abi is undefined"))?
+                .json_string()?;
+            let message = ton_sdk::Contract::construct_call_int_message_json(
+                address.clone(),
+                src_address,
+                ihr_disabled,
+                bounce,
+                value,
+                call_set.to_function_call_set(None, None, &context, &abi, true)?,
+            )
+            .map_err(|err| abi::Error::encode_run_message_failed(err, &call_set.function_name))?;
 
-        let message = ton_sdk::Contract::construct_call_int_message_json(
-            address.clone(),
-            ihr_disabled,
-            bounce,
-            CurrencyCollection::with_grams(
-                u64::from_str(&params.value)
-                    .map_err(|err| abi::Error::encode_run_message_failed(err, ""))?
-            ),
-            call_set.to_function_call_set(None, None, &context, &abi, true)?,
-        )
-        .map_err(|err| abi::Error::encode_run_message_failed(err, &call_set.function_name))?;
-
-        (message.serialized_message, address)
-    } else {
-        return Err(abi::Error::missing_required_call_set_for_encode_message());
+            (message.serialized_message, address)
+        } else {
+            let message = ton_sdk::Contract::construct_int_message_with_body(
+                address.clone(),
+                src_address,
+                ihr_disabled,
+                bounce,
+                value,
+                BuilderData::new().into()
```

Comment:
Better make body optional and pass None here. It is not obvious that message with empty body is equal to message without body

Reply:
Ok